tests/integ/test_udf.py

```diff
@@ -1448,6 +1455,97 @@ def add_one_df_pandas_annotation(df: pandas.DataFrame) -> pandas.Series:
     )
 
 
+@pytest.mark.skipif(not is_pandas_and_numpy_available, reason="pandas is required")
+@pytest.mark.parametrize(
+    "_type, data, expected_type",
+    [
+        (IntegerType, [[4096]], "object"),  # TODO: should be int16
+        (IntegerType, [[1048576]], "object"),  # TODO: should be int32
+        (IntegerType, [[8589934592]], "object"),  # TODO: should be int64
+        (FloatType, [[1.0]], "float64"),
+        (StringType, [["1"]], "string"),
+        (BooleanType, [[True]], "boolean"),
+        (BinaryType, [[(1).to_bytes(1, byteorder="big")]], "object"),
+        (DateType, [[datetime.date(2021, 12, 20)]], "datetime64[ns]"),
+        (ArrayType, [[[1]]], "object"),
+        (TimeType, [[datetime.time(1, 1, 1)]], "timedelta64[ns]"),
+        (
+            TimestampType,
+            [[datetime.datetime(2016, 3, 13, 5, tzinfo=datetime.timezone.utc)]],
+            "datetime64[ns]",
+        ),
+        (GeographyType, [["POINT(30 10)"]], "object"),
+    ],
+)
+def test_pandas_udf_input_types(session, _type, data, expected_type):
+    schema = StructType([StructField("a", _type())])
+    df = session.create_dataframe(data, schema=schema)
+
+    def return_type_in_series(x):
+        import pandas as pd
+
+        return pd.Series([x.dtype])
+
+    series_udf = udf(
+        return_type_in_series,
+        return_type=PandasSeriesType(StringType()),
+        input_types=[PandasSeriesType(_type())],
+    )
+    assert df.select(series_udf("a")).to_df("col1").collect() == [Row(expected_type)]
+
+    def return_type_in_dataframe(x):
+        return x.dtypes
+
+    dataframe_udf = udf(
+        return_type_in_dataframe,
+        return_type=PandasSeriesType(StringType()),
+        input_types=[PandasDataFrameType([_type()])],
+    )
+    assert df.select(dataframe_udf("a")).to_df("col2").collect() == [Row(expected_type)]
+
+
+@pytest.mark.skipif(not is_pandas_and_numpy_available, reason="pandas is required")
+@pytest.mark.parametrize(
+    "_type, data, expected_type",
+    [
+        (IntegerType, [[4096]], "int16"),
+        (IntegerType, [[1048576]], "int32"),
+        (IntegerType, [[8589934592]], "int64"),
+        (FloatType, [[1.0]], "float64"),
+        (StringType, [["1"]], "object"),  # TODO: should be string
```

Comment:
Is fixing this included in [SNOW-631117](https://snowflakecomputing.atlassian.net/browse/SNOW-631117)?

Reply:
nope, actually I think it's expected:
```
>>> pd.Series(["1"])
0    1
dtype: object
```
Could you check the specific type inside the series? e.g.,
```
>>> type(pd.Series(["1"])[0])
<class 'str'>
```

Reply:
You are right 👍 
[Pandas doc](https://pandas.pydata.org/docs/user_guide/text.html#text-data-types) suggests prior to pandas 1.0, `object` dtype was the only option, and it remains the default inferred type for backwards compatibility purpose.

Reply:
The doc however does recommend doing `pd.Series(["a", "b", "c"], dtype="string")`, maybe it's something we should consider doing post GA.

Reply:
good point! You can also create a jira about it for python udf team